src/corels.cpp

```diff
@@ -145,25 +152,23 @@ void evaluate_children(CacheTree* tree, Node* parent, tracking_vector<unsigned s
     }
 }
 
+static size_t num_iter = 0;
+static double min_objective = 0.0;
+static VECTOR captured, not_captured;
```

Comment:
I'd prefer for these to not be global variables -- any chance we can move them back into the function?

Reply:
I dislike how it is right now as well, but this was the first way I could think of to have a bbound_loop function that depends on the same num_iter, min_objective, etc. variables as bbound_begin and bbound_end. I suppose we could pass them as pointer arguments to those three functions instead, what do you think? Having them as static ensures they never leave corels.cpp, though.

Reply:
UPDATE: I removed the global variables from run.cpp by making them arguments. I could make these arguments as well, but then the three run functions would have a ridiculously long string of arguments (since these would have to be passed to the run functions as well as the bbound functions). What do you think?

Reply:
I think it's ok for now, but we should think about how we can incorporate them as either arguments or attributes of an object.